rfhub2/cli/client.py

```diff
@@ -0,0 +1,79 @@
+from requests import session, post, get, delete, exceptions
+from typing import Dict
+from rfhub2.model import Collection, Keyword
+
+PROTOCOL = 'http://'
+API_V1 = 'api/v1'
+TEST_COLLECTION = {"name": "a", "type": "a", "version": "a", "scope": "a",
+                   "named_args": "a", "path": "a", "doc": "a", "doc_format": "a"}
+
+
+class Client(object):
+    """
+    API client with methods to populate rfhub2 application.
+    """
+    def __init__(self, app_interface: str, port: int, user: str, password: str,):
+        self.app_interface = app_interface
+        self.port = port
+        self.auth = (user, password)
+        self.session = session()
+        self.app_url = f'{PROTOCOL}{self.app_interface}:{self.port}'
+        self.api_url = f'{self.app_url}/{API_V1}'
+
+    def check_communication_with_app(self) -> None:
+        try:
+            self.get_collections()
+        except exceptions.RequestException:
+            print(f'Connection to application at {self.app_url} refused!\n'
+                  f'Check parameters for app-interace and port.')
+            exit(1)
+        req_check_auth = self._post_request('collections', TEST_COLLECTION)
+        if req_check_auth.status_code == 201:
+            self._delete_request('collections', req_check_auth.json()["id"])
+        else:
+            print(f'{req_check_auth.reason}! Check used credentials.')
+            exit(1)
+
+    def get_collections(self) -> get:
```

Comment:
In general I would see these methods taking instances of Keyword/Collection where applicable (exact class matching those used in endpoint deifinition) and returning appropriate Keyword/Collection or list of them. If there is some error, they should raise the custom errors (should be defined in separate module) indicating to user what went wrong - connection error, not authenticated, api error, etc. with stacktrace

Reply:
If your application will not handle the error or reraise different, custom error with meaningful message it will result with application exiting with code 1 just like it does when you explicitly call `exit()`. The advantage of just raising the error is easy testing of the method raising error and not explicitly exiting the whole process. If we would want to eventually avoid displaying any stacktrace to user, i would put handling any errors with some output and exit code 1 in one place close to program main method to not have this exits scattered in various places in the code.